distributed/worker.py

```diff
@@ -900,14 +966,15 @@ def logs(self):
         return self._deque_handler.deque
 
     def log_event(self, topic, msg):
-        self.loop.add_callback(
-            self.batched_stream.send,
-            {
-                "op": "log-event",
-                "topic": topic,
-                "msg": msg,
-            },
-        )
+        full_msg = {
+            "op": "log-event",
+            "topic": topic,
+            "msg": msg,
+        }
+        if self.thread_id == threading.get_ident():
+            self.batched_stream.send(full_msg)
+        else:
+            self.loop.add_callback(self.batched_stream.send, full_msg)
```

Comment:
I'm curious. Wasn't the previous logic handling both cases?

Reply:
In the common case we're calling log_event from within the worker.  There is no reason to use add_callback.  My sense is that the thread id check is cheaper and adds less complexity than the add_callback check.

Also, I want to make sure that this signal gets added to the batched send before we start the close process.